extensions/src/main/java/org/pentaho/platform/plugin/services/email/EmailService.java

```diff
@@ -186,18 +207,50 @@ protected PasswordAuthentication getPasswordAuthentication() {
       msg.setText( messages.getString( "EmailService.MESSAGE" ) );
       msg.setHeader( "X-Mailer", "smtpsend" );
       msg.setSentDate( new Date() );
-      Transport.send( msg );
-      sendEmailMessage = "EmailTester.SUCESS";
+      sendEmail( session, msg );
+      sendEmailMessage = TEST_EMAIL_SUCCESS;
     } catch ( Exception e ) {
       logger.error( messages.getString( "EmailService.NOT_CONFIGURED" ), e );
-      sendEmailMessage = "EmailTester.FAIL";
+      sendEmailMessage = TEST_EMAIL_FAIL;
     }
     return sendEmailMessage;
   }
 
+  public void sendEmail( Session session, MimeMessage msg ) throws EmailServiceException {
+    final IEmailConfiguration emailConfig = getEmailConfig();
+    try {
+      if ( emailConfig.getAuthMechanism().equals( EmailConstants.AUTH_TYPE_XOAUTH2 ) ) {
+        IEmailAuthenticationResponse token = getOAuthToken( emailConfig );
+        if ( emailConfig.getGrantType().equals( EmailConstants.GRANT_TYPE_AUTH_CODE ) ) {
+          emailConfig.setGrantType( EmailConstants.GRANT_TYPE_REFRESH_TOKEN );
+          emailConfig.setRefreshToken( token.getRefreshToken() );
+          emailConfig.setAuthorizationCode( "" );
+          setEmailConfig( emailConfig );
+        }
+        if ( emailConfig.getSmtpProtocol().equals( EmailConstants.PROTOCOL_GRAPH_API ) ) {
+          ByteArrayOutputStream os = new ByteArrayOutputStream();
+          msg.writeTo( os );
+          String s = Base64.getEncoder().encodeToString( os.toByteArray() );
+          sendMailGraphApi( emailConfig, token.getAccessToken(), s  );
+        } else {
+          SMTPTransport transport = new SMTPTransport( session, null );
+          transport.connect( emailConfig.getSmtpHost(), emailConfig.getUserId(), null );
+          String userTokenString = String.format( USER_TOKEN_STRING, emailConfig.getUserId(), token.getAccessToken() );
+          String smtpCommand = OAUTH2_COMMAND + Base64Utils.toBase64( userTokenString.getBytes() );
+          transport.issueCommand( smtpCommand, SMTP_SC_AUTH_SUCCESS );
+          transport.sendMessage( msg, msg.getAllRecipients() );
+        }
+      } else {
+        Transport.send( msg );
```

Comment:
wouldn't we need to have `msg.getAllRecipients()` added for the case similiar the line above  `transport.sendMessage( msg, msg.getAllRecipients() );`
so ` Transport.send( msg, msg.getAllRecipients() );`